unity-environment/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs

```diff
@@ -128,8 +80,8 @@ public override void CollectObservations()
             detectableObjects = new string[] { "ball", "blueGoal", "redGoal",
                 "wall", "blueAgent", "redAgent" };
         }
-        RayPerception(rayDistance, rayAngles, detectableObjects, 0f, 0f);
-        RayPerception(rayDistance, rayAngles, detectableObjects, 1f, 1f);
+        AddVectorObs(rayPer.Percieve(rayDistance, rayAngles, detectableObjects, 0f, 0f));
```

Comment:
Can we do `AddVectorObs(rayPer)` by overloading the `AddVectorObs` ? We put the parameters in the rayPer Component.

Reply:
Not ideal for reason described above.